lib/prediction/predictor.ts

```diff
@@ -160,26 +126,81 @@ class Worker extends events.EventEmitter {
                 }));
             }
             this._requests.delete(msg.id);
-
-            if (this._minibatch.length > 0 && this._requests.size === 0)
-                this._flushRequest();
         });
     }
 
     private _failAll(error : Error) {
         for (const { reject } of this._requests.values())
             reject(error);
-        for (const ex of this._minibatch)
-            ex.reject(error);
-        this._minibatch = [];
-        this._minibatchTask = '';
-        this._minibatchStartTime = 0;
         this._requests.clear();
     }
 
-    private _flushRequest() {
+    request(task : string, minibatch : Example[]) : Promise<PredictionCandidate[][]> {
         const id = this._nextId ++;
 
+        return new Promise((resolve, reject) => {
+            this._requests.set(id, { resolve, reject });
+            //console.error(`${this._requests.size} pending requests`);
+
+            this._stream!.write({ id, task, instances: minibatch }, (err : Error | undefined | null) => {
+                if (err) {
+                    console.error(err);
+                    reject(err);
+                }
+            });
+        });
+    }
+}
+
+class RemoteWorker extends events.EventEmitter {
+    private _url : string;
+
+    constructor(url : string) {
+        super();
+        this._url = url;
+    }
+
+    start() {}
+    stop() {}
+
+    async request(task : string, minibatch : Example[]) : Promise<PredictionCandidate[][]> {
+        const response = await Tp.Helpers.Http.post(this._url, JSON.stringify({
+            id: 0, // should be ignored
+            task,
+            instances: minibatch
+        }), { dataContentType: 'application/json', accept: 'application/json' });
+        const parsed = JSON.parse(response);
+        return JSON.parse(parsed.predictions).instances.map((instance : any) : PredictionCandidate[] => {
```

Comment:
Maybe add a TODO  to remove this extra  JSON.parse() after genienlp kfserver is updated?